decidim-censuses/lib/decidim/censuses/extensions/authorize_with_age.rb

```diff
@@ -0,0 +1,48 @@
+module Decidim
+  module Censuses
+    module Extensions
+      module AuthorizeWithAge
+        def authorize
+          if authorization_handler_name == 'census_authorization_handler'
+            authorize_with_age
+          else
+            super
+          end
+        end
+
+        private
+
+        def authorize_with_age
+          return status(:missing) unless valid_metadata?
+          return status(:invalid, fields: [:birthdate]) if current_age < minimum_age
+          status(:ok)
+        end
+
+        def valid_metadata?
+          !authorization.metadata['birthdate'].nil?
+        end
+
+        def birthdate
+          @birthdate ||= Date.strptime(authorization.metadata['birthdate'], '%Y/%m/%d')
+        end
+
+        def current_age
+          now = Date.current
+          extra_year = (now.month > birthdate.month) || (
+            now.month == birthdate.month && now.day >= birthdate.day
+          )
+          now.year - birthdate.year - (extra_year ? 0 : 1)
+        end
+
+        def minimum_age
```

Comment:
When trying to convert something to an integer in Ruby it's a common practice using the .to_i object method. Every object implements it (even nil!)

It returns the integer for integers or 0 if it's not present. So, this could be an alternative implementation:

```ruby
def minimum_age
   return DEFAULT_MINIMUM_AGE if permission_options['edad'].to_i.zero?
   permission_options['edad'].to_i
end
```

However, this also has drawbacks. The number 0 does not represent a blank value, and in order to understand this piece of code you need to understand the internals of Ruby.

It's worth thinking about it. What implementation do you like more?

Reply:
I've decided to follow this way after reading at https://stackoverflow.com/questions/49274/safe-integer-parsing-in-ruby:

> you might want to watch for strings that are valid non-decimal numbers, such as those starting with 0x for hex and 0b for binary, and potentially more tricky numbers starting with zero that will be parsed as octal.

Probably too much precaution. I prefer more idiomatic ways, but I'm not 100% convinced about your solution (mostly because of two calls to `to_i` method, but probably ruby is now smart enough)... 

And by the way: `:hello.to_i` fails ;-)